### PR TITLE
[Merged by Bors] - feat: generalize tower instances for `ℚ`

### PR DESCRIPTION
**Pull request by eric-wieser**
Also add them for `ℚ≥0` by blind copy-paste.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Rat.lean [hunk 1]
```diff
@@ -44,22 +69,41 @@ theorem ratCast_smul_eq {E : Type*} (R S : Type*) [AddCommGroup E] [DivisionRing
...
+    NoZeroSMulDivisors ℕ M :=
+  ⟨fun {k} {x : M} h => by simpa [← Nat.cast_smul_eq_nsmul ℚ≥0 k x] using h⟩
+
 -- see note [lower instance priority]
 instance (priority := 100) RatModule.noZeroSMulDivisors [AddCommGroup M] [Module ℚ M] :
     NoZeroSMulDivisors ℤ M :=
-  ⟨fun {k} {x : M} h => by
-    simpa only [← Int.cast_smul_eq_zsmul ℚ k x, smul_eq_zero, Rat.zero_iff_num_zero] using h⟩
-  -- Porting note: old proof was:
-  --⟨fun {k x} h => by simpa [zsmul_eq_smul_cast ℚ k x] using h⟩
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Rat.lean:65]: This proof (modulo renamed theorems) works fine

**Review by YaelDillies — APPROVED** [review]
Amazing! Thank you very much

maintainer merge

**Comment by mattrobball** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors merge

(shake was complaining)

**Pull request closed without merging**